src/App.svelte

```diff
@@ -91,13 +91,22 @@
     if (!event) {
       return;
     }
-
+    const timeoutDuration = 2000;
     const ws = new WebSocket(relay);
     ws.onopen = () => {
       ws.send(
         JSON.stringify(["EVENT", event] satisfies Nostr.ToRelayMessage.EVENT)
       );
+      ws.onmessage = (e) => {
+        console.log(e);
+        ws.close();
+        clearTimeout(timeout); 
+      }
+      const timeout = setTimeout(() => {
+      console.log("WebSocket timeout");
       ws.close();
+    }, timeoutDuration);
+  
     };
```

Comment:
- `ws.onmessage` は最初から初期化できるのでそうしたほうが良さそうです
  - `onopen` の発火を待つ必要がないという意味です
- デバッグ用と思われる `console.log()` を消しました
- 既に閉じている `ws` をもう一度 `.close()` しても特に何も起こらないので `clearTimeout` はなくても大丈夫です

```suggestion
    ws.onopen = () => {
      ws.send(
        JSON.stringify(["EVENT", event] satisfies Nostr.ToRelayMessage.EVENT)
      );
    };
    ws.onmessage = (e) => {
      ws.close();
    }
    setTimeout(() => {
      ws.close();
    }, timeoutDuration);
```